### DISCUSSION TITLE
Turborepo

### DISCUSSION THREAD

**Discussion by multiplehats** [ricokahler/sanity-codegen#267] [General]
Hey guys!

How would one use this in a monorepo? Where `studio` is in a different folder?

**Comment by ollebergkvist** [comment]
Hi,

You can use the sanity-codegen from any dir in a monorepo, you simply define the paths to your schemas in
the sanity.config.ts file.

```
const config: SanityCodegenConfig = {
  schemaPath: '../../apps/cms/schemas',
  outputPath: './lib/sanity/types/schema.ts',
}
```


Unfortunately it appears a bug has been introduced in this module, it's currently not working from my experience.
See issue: [#266](https://github.com/ricokahler/sanity-codegen/issues/266). Hopefully the author resolves this soon.

**Reply by multiplehats** [reply]
Thank you! I'm getting a different error trying to run it in a Sveltekit project.

Maybe because it's ESM? 

Would love to use this at some point though. Saves me hours eventually!

**Reply by Yhozen** [reply]
for me having a sanity-shared (config and schemas), and a sanity studio project as well as the main project seems to work flawlessly
